kubernetes/linux/setup.sh

```diff
@@ -9,92 +9,82 @@ else
     ARCH=$1
 fi
 
-#Download utf-8 encoding capability on the ama-logs container.
-#upgrade apt to latest version
-apt-get update && apt-get install -y apt && DEBIAN_FRONTEND=noninteractive apt-get install -y locales
+sudo tdnf install ca-certificates-microsoft -y
+sudo update-ca-trust
+
+# sudo tdnf install ruby-3.1.3 -y
+tdnf install -y gcc patch bzip2 openssl-devel libyaml-devel libffi-devel readline-devel zlib-devel gdbm-devel ncurses-devel
```

Comment:
these all build dependencies, do we need all these? If we dont need, we should remove them so that we can avoid any vulnerabilities related to these packages.